dbms/src/Flash/Mpp/ReceiverChannelTryWriter.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Flash/Mpp/GRPCReceiveQueue.h>
+#include <Flash/Mpp/ReceiverChannelTryWriter.h>
+
+#include <utility>
+
+namespace DB
+{
+namespace
+{
+inline bool loopJudge(GRPCReceiveQueueRes res)
+{
+    return (res == GRPCReceiveQueueRes::OK || res == GRPCReceiveQueueRes::FULL);
+}
+
+// result can not be changed from FULL to OK
+inline void updateResult(GRPCReceiveQueueRes & dst, GRPCReceiveQueueRes & src)
+{
+    if (unlikely(src != GRPCReceiveQueueRes::OK))
```

Comment:
For `{channelA, channelB, channelC}`, it's possible for `channelA` and `channelC` to return `OK`, and `channelB` to return `FULL`, it's just not allowed to set the final result to `channelC`'s `OK` right?

Reply:
If it's just not allowed to update the result, rather than this situation being impossible, "unlikely" shouldn't be used.

Reply:
> For `{channelA, channelB, channelC}`, it's possible for `channelA` and `channelC` to return `OK`, and `channelB` to return `FULL`, it's just not allowed to set the final result to `channelC`'s `OK` right?

yes

Reply:
> t's just not allowe

okk